tasks/Module 4/task_2.py

```diff
@@ -0,0 +1,31 @@
+"""Fill in the Gaps."""
+import os
+
+
+def fill_gaps(folder_path, prefix):
+    """Fill in the Gaps."""
+    # init file list with first not needed element
+    file_list = [None]
+    os.chdir(folder_path)
+    # get list of tuples of filenames and extentions
+    with os.scandir(folder_path) as scandir:
+        for entry in scandir:
+            if entry.name.startswith(prefix) and entry.is_file():
+                file_list.append(os.path.splitext(entry.name))
+    # check if sequence number equal to number in filename and rename it if is not.
+    number_of_zeroes = len(file_list[1][0]) - len(prefix)
+    for i in range(1, len(file_list)):
+        number = int(file_list[i][0][len(prefix):])
+        if i != number:
+            new_file_name = prefix + str(i).zfill(number_of_zeroes) \
+                            + file_list[i][1]
+            os.renames(''.join(file_list[i]), new_file_name)
+            print(new_file_name)
+
+
+if __name__ == '__main__':
+    userprofile = os.environ.get('USERPROFILE') \
```

Comment:
If I understand correctly you wanted to ensure that no matter if you call this script from inside or outside `Module 4` directory it will look for files in this directory. You can make use of `__file__` special variable to avoid hardcoding the path. This variable is defined implicitly in every Python file and contains absolute patch of that file. Thanks to this you can use `os.path.dirname(__file__)` to get absolute path to the directory where is your script.

Even better would be to pass path as command line argument. The simplest (not the best) way is to use [sys.argv]( (https://docs.python.org/3/library/sys.html#sys.argv):
```python
import sys

try:
    input_directory = sys.argv[1]
except IndexError:
    print('No directory provided, using current directory')
    input_directory = '.'

fill_gaps(input_directory)
```

Reply:
Thank you for great hint with __file__. 
Somehow when i tried to do `op.path.join(os.environ.get('USERPROFILE'), 'some_dir')`
it returned "C:\some_dir"
but, when i `print(os.environ.get('USERPROFILE'))` it printed full path to my home dir.

Reply:
Weird. I haven't got a Windows machine to try, but you can also try with `pathlib`: `pathlib.Patch.home() / 'some_dir'`.